### PR TITLE
Elg 267

### PR DESCRIPTION
**Pull request by gcleary**
Addressed the device_id length issue and found a number of other issues related to AES_KEYLEN and include files. Release notes should now mention that the client code (and sample_client.c) no longer should set SKY_LIBEL which is reserved  for the internal libel files.

### CODE REVIEW

File: libel/libel.h [hunk 1]
```diff
@@ -132,8 +131,12 @@ typedef int (*Sky_randfn_t)(uint8_t *rand_buf, uint32_t bufsize);
 typedef time_t (*Sky_timefn_t)(time_t *t);
 
 #ifndef SKY_LIBEL
+#include "aes.h"
+#include "crc32.h"
 typedef void Sky_ctx_t;
+#define MAC_SIZE 6
 #else
+#include "aes.h"
```
Comment (tboinske-skyhook) [inline libel/libel.h:139]: This file is included in both branches of the ifndef conditional. Why not just include outside the conditional altogether?

Comment (gcleary) [reply libel/libel.h]: Duh! Yup, done.

File: libel/libel.h [hunk 2]
```diff
@@ -132,7 +132,9 @@ typedef int (*Sky_randfn_t)(uint8_t *rand_buf, uint32_t bufsize);
 typedef time_t (*Sky_timefn_t)(time_t *t);
 
 #ifndef SKY_LIBEL
+#include "crc32.h"
```
Comment (tboinske-skyhook) [inline libel/libel.h:135]: Why is this needed in this case?

Comment (tboinske-skyhook) [reply libel/libel.h:135]: And anyway, if it's needed, it too can be factored out of the conditional statement.

Comment (gcleary) [reply libel/libel.h:135]: The sample_client includes an example of how to validate the saved context and extract the size from the header. Rather than expecting our users to include crc32.h along with libel.h, seems reasonable to provide these definitions when the user includes libel.h.
Yes, I duplicated it, and it is better to factor it out.

Comment (tboinske-skyhook) [inline libel/libel.h]: I suggest reversing the sense of the conditional. "if ..." reads a little more easily than "if not...".
